src/main/kotlin/no/item/xp/plugin/util/NodeParserHelpers.kt

```diff
@@ -0,0 +1,20 @@
+package no.item.xp.plugin.util
+
+import arrow.core.Option
+import no.item.xp.plugin.extensions.*
+import org.w3c.dom.Node
+
+fun getCommentForGeneratedField(node: Node): Option<String> =
+  node.getChildNodeAtXPath(getXpathExpressionFromString("//label"))
+    .flatMap(Node::getTextContentAsOption)
+
+fun getNameForGeneratedField(node: Node): Option<String> =
+  node.getAttributeAsOption("name")
+
+fun mapToString(node: Node): Sequence<String> {
+  val seq: Sequence<String> = emptySequence()
+  node.getChildNodesSequence().forEach {
+    seq.plus(it.textContent.toString())
+  }
+  return seq
```

Comment:
Never use `forEach` since it forces us to either do a side-effect or mutate something, use `map` instead.

```suggestion
  return node
    .getChildNodesSequence()
    .map { it.textContent.toString() }
```